packages/eslint-plugin-pf-codemods/src/rules/v6/chipReplaceWithLabel/chip-replace-with-label.ts

```diff
@@ -0,0 +1,202 @@
+    (specifier) =>
+      specifier.type === 'ImportSpecifier' &&
+      specifier.imported.name === specifierName
+  ) as ImportSpecifier | undefined;
+};
+
+const renameImportSpecifier = (
+  importToRename: ImportSpecifier | undefined,
+  node: ImportDeclaration,
+  fixer: Rule.RuleFixer,
+  newName: string
+) => {
+  if (!importToRename) {
+    return [];
+  }
+
+  const oldName = importToRename.imported.name;
+  const importSpecifier = getImportSpecifier(
+    node,
+    importToRename.imported.name
+  ) as ImportSpecifier;
+
+  const importedName = importSpecifier.imported.name;
+  const name = importedName.replace(oldName, newName);
+  const localName = importSpecifier.local.name;
+  const aliasText = importedName !== localName ? ` as ${localName}` : '';
+  const rename = `${name}${aliasText}`;
+  return [
+    fixer.replaceText(importSpecifier, newName),
+    fixer.replaceText(node.source, "'@patternfly/react-core'"),
+  ];
+};
+
+const getJSXAttribute = (node: JSXElement, attributeName: string) => {
+  return node.openingElement.attributes.find(
+    (attribute) =>
+      attribute.type === 'JSXAttribute' && attribute.name.name === attributeName
+  ) as JSXAttribute | undefined;
+};
+
+const parseBadgeAttributeValue = (
+  badgeAttributeValue: JSXAttributeValue
+): Node => {
+  if (badgeAttributeValue.type !== 'JSXExpressionContainer') {
+    return badgeAttributeValue;
+  }
+
+  const isValueJSX = ['JSXElement', 'JSXEmptyExpression'].includes(
+    badgeAttributeValue.expression.type
+  );
+
+  return isValueJSX ? badgeAttributeValue.expression : badgeAttributeValue;
+};
+
+const moveBadgeAttributeToBody = (
+  node: JSXElement,
+  fixer: Rule.RuleFixer,
+  context: Rule.RuleContext
+) => {
+  const badgeAttribute = getJSXAttribute(node, 'badge');
+
+  return badgeAttribute
+    ? [
+        fixer.insertTextBefore(
+          node.closingElement as JSXClosingElement,
+          context
+            .getSourceCode()
+            .getText(
+              badgeAttribute?.value
+                ? parseBadgeAttributeValue(badgeAttribute.value)
+                : undefined
+            )
+        ),
+        fixer.remove(badgeAttribute),
+      ]
+    : [];
+};
+
+const renameOnClickAttribute = (node: JSXElement, fixer: Rule.RuleFixer) => {
+  const onClickAttribute = getJSXAttribute(node, 'onClick');
+
+  return onClickAttribute
+    ? [fixer.replaceText(onClickAttribute.name, 'onClose')]
+    : [];
+};
+
+// https://github.com/patternfly/patternfly-react/pull/10049
+module.exports = {
+  meta: { fixable: 'code' },
+  create: function (context: Rule.RuleContext) {
+    const { imports } = getFromPackage(
+      context,
+      '@patternfly/react-core/deprecated'
+    );
+
+    const chipImport = imports.find(
+      (specifier) => specifier.imported.name === 'Chip'
+    );
+
+    const chipGroupImport = imports.find(
+      (specifier) => specifier.imported.name === 'ChipGroup'
+    );
+
+    return !chipImport && !chipGroupImport
```

Comment:
Nit: I would reverse this conditional so that these expressions aren't notted and have the non-empty object returned first